### PR TITLE
Don't let users watch Web Sessions with secrets

### PR DESCRIPTION
**Pull request by Joerger**
Users are currently permitted to set up watchers on their own web sessions. This is used by clients to check when a web session is propagated through the cache and ready to support connections. In these cases the user does not actually need the web session secrets. At most they need the web session ID.

This PR adds logic for `kind.LoadSecrets` when watching web sessions, and prevents users from watching their own secrets. This is particularly important for Hardware Key support as the expectation is that web session secrets are only [known by the Proxy and Auth services](https://github.com/gravitational/teleport/blob/master/rfd/0080-hardware-key-support.md#web-session-access).

Note: This is not really a security fix as an attacker would still need access to the user's hardware key and MFA device to carry out an attack. This just hardens Hardware Key support for the cluster in general.

Changelog: Harden web session security for Hardware Key support

### CODE REVIEW

**Comment by Joerger** [comment]
@tcsc @fspmarshall friendly reminder to review

**Review by fspmarshall — APPROVED** [review]
This needs backwards-compatibility logic to ensure that auth servers running this logic don't break older proxies.  If an auth receives read/watch requests from older proxies it should modify the request inflight to set `loadSecrets` to true.  LGTM otherwise.

File: api/types/session.go [hunk 1]
```diff
@@ -86,8 +85,8 @@ type WebSession interface {
 	SetLoginTime(time.Time)
 	// GetIdleTimeout returns the max time a user can be inactive for this session.
 	GetIdleTimeout() time.Duration
-	// WithoutSecrets returns copy of the web session but without private keys
-	WithoutSecrets() WebSession
+	// CopyWithoutSecrets returns copy of the web session but without private keys
+	CopyWithoutSecrets() WebSession
```
[the remark this answers is not in the document]
Comment (Joerger) [reply api/types/session.go]: addressed in 7e0a738a77108407745943e482156aa8f6dc2cbf

File: api/types/session.go [hunk 2]
```diff
@@ -194,18 +193,11 @@ func (ws *WebSessionV2) GetIdleTimeout() time.Duration {
...
-	ws.Spec.LoginTime = ws.Spec.LoginTime.UTC()
-	ws.Spec.BearerTokenExpires = ws.Spec.BearerTokenExpires.UTC()
-
-	cp := proto.Clone(ws).(*WebSessionV2)
+	cp := *ws
 	cp.Spec.Priv = nil
 	cp.Spec.SAMLSession = nil
 	cp.Spec.DeviceWebToken = nil
-	return cp
+	return &cp
```
[the remark this answers is not in the document]
Comment (Joerger) [reply api/types/session.go:200]: No I think this was an uncaught merge conflict, thanks for catching it.

[the remark this answers is not in the document]
Comment (Joerger) [reply api/types/session.go]: Ah no I remember now, it was causing race conditions in tests when the times are changed to UTC, which is needed to make proto.Clone work with gogoproto.

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**